### PR TITLE
Define Firefox versions for several css files

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
This is to help the efforts of #3126 by defining Firefox versions for many CSS entries.  Looking through the Git mirror of `mozilla-central` (`gecko-dev`), I've found the following commits where the CSS rules originally appear.

# Found and Versioned

## Exact Commits of Implementation
These are the earliest commits that add support for the feature.

**css.at-rules.import**
```
commit f7975c0847426f13048aea235b3d62f0dc773e80
Author: peterl%netscape.com <peterl%netscape.com>
Date:   Thu Jun 10 05:32:38 1999 +0000
```

**css.at-rules.media.color, css.at-rules.media.device-aspect-ratio, css.at-rules.media.device-height, css.at-rules.media.device-width, css.at-rules.media.grid, css.at-rules.media.monochrome, css.at-rules.media.orientation, css.at-rules.media.height, css.at-rules.media.width**
```
commit 62b65d284559c134b89c16aa4e8aed7f54064877
https://hg.mozilla.org/mozilla-central/rev/6298e0ab72f2
Author: L. David Baron <dbaron@dbaron.org>
Date:   Sat Jul 26 09:14:48 2008 -0700
```

## Evidence of Use in 1.x
These commits show the properties being used in Firefox 1.x.

**css.properties.-moz-image-region**
```
commit 8eb4e1f367c20d0539d7aab6feecb3cf056f4052
https://hg.mozilla.org/mozilla-central/rev/24ad1b4bc1f2
Author: kairo@kairo.at <kairo@kairo.at>
Date:   Fri Aug 10 03:43:28 2007 -0700
```

**css.properties.-moz-user-focus**
```
commit 87e005b40bd4fdd1ae3f6b1df5b61a49703546d5
https://hg.mozilla.org/mozilla-central/rev/5eafaafb7491
Author: Gavin Sharp <gavin.sharp@gmail.com>
Date:   Mon Jun 15 15:43:34 2009 +0200
```

# Not In Compat Table
- css.properties.background-image.image-rect
- css.properties.background-image.any_image

# In Progress
These have commits that show their implementation in Servo, but not their implementation in Firefox altogether.  It's a starting point for some parts, though.

**css.properties.-moz-user-input, css.properties.-moz-user-input-none, css.properties.-moz-user-input.enabled, css.properties.-moz-user-input.disabled**
```
commit 4df4e0e73e6dc50619bdf1914346792d445ad178
https://hg.mozilla.org/mozilla-central/rev/86cd364d1229
Author: Keith Yeung <kungfukeith11@gmail.com>
Date:   Tue May 17 14:57:32 2016 -0700

Push ID 31307
https://hg.mozilla.org/mozilla-central/pushloghtml?changeset=c3a5180a727e
https://hg.mozilla.org/mozilla-central/rev/94079d43835f
```

**css.properties.-moz-user-input.auto**
```
commit c374fd6e7e8d7770474e65c69a1bece9238b90a7
https://hg.mozilla.org/mozilla-central/rev/c3a5180a727e
Author: Xidorn Quan <me@upsuper.org>
Date:   Wed Jan 18 01:22:00 2017 -0800

Push ID 31307
https://hg.mozilla.org/mozilla-central/pushloghtml?changeset=c3a5180a727e
https://hg.mozilla.org/mozilla-central/rev/94079d43835f
```

**css.properties.align-content.flex_context.start_end, css.properties.align-content.flex_context.baseline, css.properties.align-self.flex_context.start_end, css.properties.align-self.flex_context.baseline**
```
commit 286dc70695bad39883f9af88eff899a152ccbbc6
https://hg.mozilla.org/mozilla-central/rev/5e36f837a50a
Author: Pu Xingyu <pu.stshine@gmail.com>
Date:   Tue May 31 21:44:06 2016 -0500

Push ID 31307
https://hg.mozilla.org/mozilla-central/pushloghtml?changeset=c3a5180a727e
https://hg.mozilla.org/mozilla-central/rev/94079d43835f
```

# To Do
- css.properties.border-radius.4_values_for_4_corners
- css.properties.caption-side.non_standard_values
- css.properties.contain
- css.properties.cursor.url_positioning_syntax
- css.properties.justify-content.flex_context.start_end
- css.properties.justify-content.flex_context.baseline
- css.properties.mask
- css.properties.outline-color.invert
- css.properties.place-content.flex_context
- css.properties.text-rendering.auto
- css.properties.text-rendering.geometricPrecision
- css.properties.transition-property.IDENT_value
- css.properties.user-modify
- css.properties.user-select.auto
- css.properties.user-select.all
- css.properties.user-select.text
- css.properties.visibility.collapse
- css.selectors.-moz-color-swatch
- css.selectors.-moz-page-sequence
- css.selectors.-moz-page
- css.selectors.-moz-range-progress
- css.selectors.-moz-range-thumb
- css.selectors.-moz-range-track
- css.selectors.-moz-scrolled-page-sequence
- css.selectors.blank
- css.selectors.indeterminate
- css.selectors.placeholder-shown.non_text_types
- css.selectors.read-only
- css.selectors.read-write
- css.selectors.read-write.matches_editable_elements
- css.types.image.gradient
- css.types.image.gradient.linear-gradient
- css.types.length.ex

### CODE REVIEW

**Comment by ddbeck** [comment]
This is really great @vinyldarkscratch! A question for you, mostly for my own edification: how did you sync up the commits to a release (especially the version 54 items)?

**Comment by vinyldarkscratch** [comment]
For the first one, `css.at-rules.import`, since we started getting into commits from 1999, I figured that it's obviously going to be a Firefox 1.0 (since Phoenix 0.1 launched in late 2002).  The rest of them, I looked at their associated milestone.  (I hope that I'm doing it right!)

**Comment by vinyldarkscratch** [comment]
After thinking about it though, and whilst looking into `css.properties.border-radius.4_values_for_4_corners`, I realize that I was looking at the [Servo](https://github.com/servo/servo) engine, which was created on [February 8, 2012](https://api.github.com/repos/servo/servo), and not looking through Firefox as a whole.  Many of the properties could be existent in versions older than Servo.  I'll need to do some more research.

**Comment by vinyldarkscratch** [comment]
@ddbeck I actually reverted all the commits that set the versions to when it was added in Servo, so everything that's in this PR is now all ready to go, preferably after a quick review!

File: css/properties/-moz-image-region.json [hunk 1]
```diff
@@ -18,10 +18,10 @@
               "version_added": false
             },
             "firefox": {
-              "version_added": true
+              "version_added": "54"
```
Comment (ddbeck) [inline css/properties/-moz-image-region.json]: I'm still unclear on these two. The PR description suggests an early version (1/4), not 54

Comment (vinyldarkscratch) [reply css/properties/-moz-image-region.json]: ...oops.  That was meant to be set to “1”.  Fixing now!

**Comment by vinyldarkscratch** [comment]
Thank **you!**  And don't worry, this is only the beginning of a wave of updates related to the issue.  :wink:

**Pull request merged**